### DISCUSSION TITLE
how do  this  instant-ngp deal with  nerf_llff_data?

### DISCUSSION THREAD

**Discussion by lishuwei0424** [NVlabs/instant-ngp#367] [Q&A]
how do  this  instant-ngp deal with  nerf_llff_data? thanks you

**Comment by ashawkey** [comment]
Hi, I wrote a script to convert the LLFF format to NeRF-compatible format, by generating `transforms.json` from `poses_bounds.npy`:
https://github.com/ashawkey/torch-ngp/blob/main/llff2nerf.py
However, it doesn't convert rays to the NDC space, but more like running `colmap2nerf.py` on the original images.
The results for the `fern`:
![Snipaste_2022-04-14_10-00-27](https://user-images.githubusercontent.com/25863658/163300524-bf243109-01aa-4b95-9736-62195bcc50de.jpg)

**Reply by kwea123** [reply]
How is the performance?

**Reply by ashawkey** [reply]
I have not tested the exact performance. Honestly, this may not be a good solution, since the script fails for the `flower` scene (the cameras are almost parallel, so the script cannot estimate the scene center?), and needs a manual fix.

**Reply by qingxu-thu** [reply]
I've checked llff2nerf.py in your project and found it corresponding to the colmap2nerf and llff's implementation. When I try to take it into the real scenes, it fails (but the scene seems to have a little large FOV..) with many floaters. Hope to contribute a bad case for you to analyze what happened in such scenes...

**Comment by kwea123** [comment]
With a different coordinate conversion, my [ngp_pl](https://github.com/kwea123/ngp_pl) can now train successfully on every `llff` scene (including flower that doesn't work here). However the result is worse than original NeRF, possibly because here we don't use NDC, but with the simplified code I think it's worth a try.

![000](https://user-images.githubusercontent.com/11364490/180428510-ddc70aa8-926e-4e88-b20a-8d1284b86d9f.png)
![000_d](https://user-images.githubusercontent.com/11364490/180428519-a680983b-a761-462a-a6ad-df267dc0cdc1.png)

**Reply by AIBluefisher** [reply]
Is this a validation image or training image? I found that using the provided dataloader the training is normal while the validation psnr is very low.
![image](https://user-images.githubusercontent.com/18096136/218106014-d91e27e2-57f5-4027-8dd3-44f3e378815c.png)
![image](https://user-images.githubusercontent.com/18096136/218106059-07a3a716-19b9-4cd8-93de-d7f739709aa9.png)

**Reply by AIBluefisher** [reply]
> Is this a validation image or training image? I found that using the provided dataloader the training is normal while the validation psnr is very low. ![image](https://user-images.githubusercontent.com/18096136/218106014-d91e27e2-57f5-4027-8dd3-44f3e378815c.png) ![image](https://user-images.githubusercontent.com/18096136/218106059-07a3a716-19b9-4cd8-93de-d7f739709aa9.png)

I used nerfacc as my pipeline. Not sure if npg_pl has the same issue. See a related issue from nerfacc [here](https://github.com/KAIR-BAIR/nerfacc/issues/120)
